copy/copy.go

```diff
@@ -489,8 +497,20 @@ func (ic *imageCopier) copyBlobFromStream(srcStream io.Reader, srcInfo types.Blo
 		inputInfo.Size = -1
 	}
 
+	var stream io.Reader
+
 	// === Finally, send the layer stream to dest.
-	uploadedInfo, err := ic.dest.PutBlob(destStream, inputInfo)
+	// the pipe is here to facilitate the stream copier which will act as an
+	// intermediary to signal progress information to any registered channel.
+	if ic.signal != nil && ic.signalInterval > 0 {
+		r, w := io.Pipe()
+		go streamcopy.WithProgress(w, destStream, inputInfo, ic.signalInterval, ic.signal)
```

Comment:
Shouldn’t this use `srcInfo` instead of `inputInfo`? `inputInfo` may be `{Digest: "", Size:-1}` in the “compressing on the fly” case.